tests/system_tests/test_misc.py

```diff
@@ -22,3 +22,45 @@ def test_with_no_output(self):
         cwd = os.path.dirname(filename)
         self.run_test_with_no_output(
             DebugInfo(filename=filename, cwd=cwd))
+
+
+class ThreadCountTests(LifecycleTestsBase):
+    def run_test_threads(self, debug_info, bp_filename, bp_line, count):
+        breakpoints = [{
+            'source': {
+                'path': bp_filename
+            },
+            'breakpoints': [{
+                'line': bp_line
+            }]
+        }]
+        with self.start_debugging(debug_info) as dbg:
+            session = dbg.session
+            with session.wait_for_event('stopped') as result:
+                lifecycle_handshake(
+                                    session, debug_info.starttype,
+                                    breakpoints=breakpoints,
+                                    threads=True)
+            event = result['msg']
```

Comment:
You might want to add a delay of 1s, cuz today when you add a break point not all thread stop. So a 1s delay will ensure other thread to have hot the breakpoint and stopped.